### ISSUE TITLE
ENH: Pipe for transformer-based Regression

### ISSUE THREAD

**Issue by GevinHasmitha** [shap/shap#3963]
### Problem Description

1

I'm developing a transformer-based regression model using a classification model with num labels set to 1 so that it can be used for regression.

```
from transformers import AutoTokenizer, AutoModelForSequenceClassification, DataCollatorWithPadding
from torch.utils.data import DataLoader

BASE_MODEL = "distilbert/distilbert-base-uncased"
LEARNING_RATE = 2e-5
MAX_LENGTH = 512
BATCH_SIZE = 16
EPOCHS = 20

tokenizer = AutoTokenizer.from_pretrained(BASE_MODEL)
model = AutoModelForSequenceClassification.from_pretrained(BASE_MODEL, num_labels=1)
```
This model will be trained on my dataset which consists of text and respective scores as shown below e.g.;

```
[
    {
        "text": "some text",
        "score": "7"
    },
    {
        "text": "some text",
        "score": "8"
    }
]
```
Is there a pipe available for such regression purposes? If not, will it be implemented in the future?

### Feature request checklist

- [x] I have checked the issue tracker for duplicate issues.
- [ ] I'd be interested in making a PR to implement this feature
